matrix/matrix_binary_search.py

```diff
@@ -0,0 +1,47 @@
+def binary_search(array: list, lower_bound: int, upper_bound: int, value: int) -> bool:
+    """
+    This function carries out Binary search on a 1d array
+    Array: A 1d sorted array
+    lower_bound: the lowest index for the binary search algorithm
+    upper_bound: the highest index for the binary search algorithm
+    value : the value ment to be searched
+    >>> matrix = [1, 4, 7, 11, 15]
+    >>> target = 1
+    >>> binary_search(matrix, 0, len(matrix) - 1, 1)
+    True
+    >>> target = 23
+    >>> binary_search(matrix, 0, len(matrix) - 1, 1)
+    False
+    """
+    r = int((lower_bound + upper_bound) // 2)
+    if array[r] == value:
+        return True
+    if r in (lower_bound, upper_bound):
+        return False
+    if array[r] < value:
+        return binary_search(array, r + 1, upper_bound, value)
+    else:
+        return binary_search(array, lower_bound, r, value)
+def mat_bin_search(value: int, matrix: list) -> bool:

```

Comment:
You NEED two blank lines before and after a function of psf/black will fail your pull request.  See CONTRIBUTING.md
```suggestion


def mat_bin_search(value: int, matrix: list) -> bool:
```